auth/src/main/java/org/aerogear/auth/AuthService.java

```diff
@@ -0,0 +1,68 @@
+package org.aerogear.auth;
+
+import org.aerogear.auth.credentials.ICredential;
+import org.aerogear.auth.impl.AuthenticatorFactory;
+
+import java.security.Principal;
+import java.util.concurrent.Future;
+
+/**
+ * Entry point for authenticating users.
+ */
+public class AuthService {
+
+    /**
+     * Authentication service singleton.
+     */
+    private static AuthService INSTANCE;
+
+    private final AuthenticatorFactory authenticatorFactory;
+
+    /**
+     * Instantiates a new AuthService object
+     * @param config Authentication Service configuration
+     */
+    private AuthService(final AuthServiceConfig config) {
+        this.authenticatorFactory = new AuthenticatorFactory(config);
+    }
+
+    /**
+     * Log in the user with the given credential. Flow to be used to authenticate the user is automatically
+     * selected by analysing the received credentials. If the credentials are null,
+     * the browser will be open asking for authentication
+     *
+     * The login will be asynchronous.
+     *
+     * @param username The username
+     * @param credentials the credential
+     * @return a user principal
+     */
+    public Future<Principal> login(final String username, final ICredential credentials) {
```

Comment:
I don't think `username` is required as a separate parameter here. It can be included in the implementation class of the `ICredential` interface if it's required. The credential type we are going to support doesn't require `username`